### ISSUE TITLE
Lazy "slider" for plotting large tables

### ISSUE THREAD

**Issue by devonpallison** [deephaven/deephaven-plugins#452]
As a user, I want a way plot a large table so that the only part of my table is loaded into memory at a time. For example, a large time series plot with a range slider on the x axis that "lazily" loads in the table data as I scroll.

**Comment by mattrunyon** [comment]
How would this data/table be stored and lazy loaded on the server? We could probably do something like require the x-axis column is sorted and then use the `seekRow` API to find the table viewport range we want to subscribe to based on the range. But I'm guessing that might end up loading the whole table for the search

**Comment by mofojed** [comment]
@devonpallison Downsampling will handle viewporting data for large line time-series plots. Is this for a different plot type, or is there something with downsampling that wouldn't achieve this? Would like some more clarification.

**Comment by devonpallison** [comment]
> @devonpallison Downsampling will handle viewporting data for large line time-series plots. Is this for a different plot type, or is there something with downsampling that wouldn't achieve this? Would like some more clarification.

Are you saying that this is already handled for line plots? It would be nice to have for other styles as well, but the user requesting this asked for a line plot.

**Comment by dsmmcken** [comment]
> Are you saying that this is already handled for line plots?

Depends on which plotting code we are talking about, yes in legacy, not yet in dx.

What plotting code are we talking here? Like what line plot command is the user running?

**Comment by mattrunyon** [comment]
Pretty sure downsampling with a range does not mean the table isn't fully loaded on the server. It just adjusts the downsampling so it's accurate over the specified range.

**Comment by dsmmcken** [comment]
> legacy = deephaven.plot.figure and dx = plotly express 

Yes. Specifically though `dx` is our wrapper around  plotly express (`px`) which allows you to pass it ticking table, and not just a pandas df.  (`from deephaven.plot import express as dx`)

Legacy already does automatic down sampling of timeseries plots for line plots. And when you zoom/select a specific range on that plot, it also viewports it to just that range. 

Down sampling is in progress for dx still.

**Comment by vbabich** [comment]
Sounds like downsampling should solve this.
